lms/djangoapps/courseware/access.py

```diff
@@ -562,6 +565,20 @@ def can_load():
         if not group_access_response:
             return group_access_response
 
+        if is_integrity_signature_enabled():
+            # If the user's enrollment mode is non-audit, check if they have an integrity
+            # signature for the course, and deny access if not
+            course_id = str(course_key)
+            enrollment = get_enrollment(user.username, course_id)
+            if enrollment and enrollment.get('mode') != 'audit':
```

Comment:
I'm not sure if this is needed. On line 565, group access is already checked which accounts for FBE, and will early return if a user does not have access due to their enrollment group.